security/pkg/pki/ca/ca.go

```diff
@@ -383,3 +389,28 @@ func (ca *IstioCA) GenKeyCert(hostnames []string, certTTL time.Duration) ([]byte
 
 	return certPEM, privPEM, nil
 }
+
+// if CA cert becomes invalid before workload cert it's going to cause workload cert to be invalid too,
```

Comment:
This comment should go to the place where this function is being called, and why do we change the function name? It is better to be called CertMinTTL (or something similar to this)?

Note the logic is that this is a helper function that could be called in many places, its name and comment should be focusing on the function itself instead of explaining why it's called for a specific place, which should go to the original calling site.

Reply:
I moved the comment to where it's called.

About the function name, I've changed a few times already, could we agree with a final name for this function, please?

@lei-tang

Reply:
The function name getDefaultCertTTL and the argument defaultCertTTL is a bit confusing of what is actually doing, the `default` is a super overloaded term that you don't really know what does it mean without looking into the code, and it's more commonly used for setting some hard-coded static value.

Here we're dynamically getting the minTTL between certChainPem and defaultCertTTL, so minTTL(certChainPem []byte, defaultCertTTL time.Duration) sounds better as it's super clear about what it is doing by just looking at the function name.

Reply:
thanks, the PR looks good to me and if there are concerns of the name from https://github.com/istio/istio/pull/25158#discussion_r450564614, you can  use a more specific name like minWorkloadCertTTL().

 I will defer to @lei-tang for the final approval in case lei has more comments.